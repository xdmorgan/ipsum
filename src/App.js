import React, { Component } from "react";
import styled from "styled-components/primitives";
import logo from "./logo.svg";

const AppView = styled.View`
  display: flex;
  align-items: center;
`;
const HeaderView = styled.View`
  background-color: #222;
  height: 150px;
  padding: 20px;
`;

const LogoImage = styled.Image`
  height: 80px;
`;

const IntroHeading = styled.Text`
  font-size: 1.5em;
  color: #fff;
`;

const IntroParagraph = styled.Text`
  font-size: large;
`;

class App extends Component {
  render() {
    return (
      <AppView>
        <HeaderView>
          <LogoImage
            source={logo}
            style={{ resizeMode: "contain" }}
            alt="logo"
          />
          <IntroHeading>Welcome to React</IntroHeading>
        </HeaderView>
        <IntroParagraph>
          To get started, edit <code>src/App.js</code> and save to reload.
        </IntroParagraph>
      </AppView>
    );
  }
}

export default App;
