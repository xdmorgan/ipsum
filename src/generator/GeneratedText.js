import React from "react";
import styled from "styled-components/primitives";
import { generate } from "./";

const StyledTextView = styled.View`
  padding: 20px 0;
`;
const StyledText = styled.Text`
  color: #111;
  font-size: 18px
  line-height: ${18 * 1.4}px;
`;

export default class GeneratedText extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      content: generate(props.data, props.length)
    };
  }
  render() {
    return (
      <StyledTextView>
        <StyledText>{this.state.content}</StyledText>
      </StyledTextView>
    );
  }
}
