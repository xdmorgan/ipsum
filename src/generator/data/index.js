const NUFC: string[] = [
  "Rafas mags",
  "Shola on a bike eating an apple",
  "Mandy's mags",
  "Sissoko's debut against Chelsea",
  "Rafa putting his glasses in his pocket, job done lads!",
  "Pardew headbutting David Meyler",
  "Jonas goal vs West Ham",
  "Yohan Kebabs",
  "Ben Arfa's goal against Everton",
  "Ben Arfa's goal against Blackburn",
  "Tiote's goal against Arsenal",
  "Dwight Goal",
  "Geordie boys"
];

export { NUFC };
