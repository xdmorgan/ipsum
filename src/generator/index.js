// @flow

/**
 * Return a random entry from the supplied collection. Use js default
 * Math.random() unless another randomizer function is provided.
 * @param {*} arr An arbitrary collection of whatever
 * @param {*} rand A randomizer function (returns 0-1)
 */
const randomEntryInArray = (
  arr: any[] = [],
  rand: () => number = Math.random
) => arr[Math.floor(rand() * (arr.length - 1))];

/**
 *
 * @param {*} dict A dictionary of strings to select from
 * @param {*} length How many entries to randomly select
 */
const randomCollection = (dict: string[] = [], length: number = 100) =>
  Array.from({ length }).map(() => randomEntryInArray(dict));

/**
 *
 * @param {*} data A dictionary of strings to select from
 * @param {*} length How many entries to randomly select
 */
const generate = (data: string[], length: number) =>
  randomCollection(data, length).join(". ");

export { default as GeneratedText } from "./GeneratedText";
export { randomEntryInArray, randomCollection, generate };
export default generate;
