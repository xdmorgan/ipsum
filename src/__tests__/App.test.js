import React from "react";
import renderer from "react-test-renderer";
import App from "../App";
import "jest-styled-components";

import * as generator from "../generator";
generator.generate = () => "Randomly generated String";

it("renders without crashing", () => {
  const tree = renderer.create(<App />);
  expect(tree).toMatchSnapshot();
});
