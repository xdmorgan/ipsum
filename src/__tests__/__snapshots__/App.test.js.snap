// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
>
  <div
    className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-1m2hcu5 rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
  >
    PersistentHeader
  </div>
  <div
    className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-maxWidth-11niif6 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingBottom-1mdbw0j rn-paddingLeft-1vwolk8 rn-paddingRight-rnv2vh rn-paddingTop-wk8lta rn-position-bnwqim rn-width-13qz1uu rn-zIndex-1lgpqti"
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1ui5ee8 rn-fontWeight-b88u0q rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      Lorem Ipsum Generator
    </div>
    <div
      className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingBottom-k8qxaj rn-paddingLeft-gy4na3 rn-paddingRight-9aemit rn-paddingTop-1knelpx rn-position-bnwqim rn-zIndex-1lgpqti"
    >
      <div
        className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-hu3udz rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1i10wst rn-lineHeight-eqdxtu rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
        dir="auto"
      >
        Randomly generated String
      </div>
    </div>
  </div>
</div>
`;
